### ISSUE TITLE
Variables defined in environment cannot have trailing spaces

### ISSUE THREAD

**Issue by ColemanTom** [cylc/cylc-flow#5080]
**Describe the bug**

If you define a variable like follows, the result in the job script is `_=""`. I would expect it to be `_=" "`.

```
[environment]
    _ = " "
```

I posted this in the [Cylc discourse](https://cylc.discourse.group/t/define-a-variable-in-the-environment-section-which-ends-in-a-space/521)

NOTE: Depending on what you are doing, you could work around it by defining `_ = " \0"`, but that isn't ok in all situations (it was in mine).

**Release version(s) and/or repository branch(es) affected?**

7.8.4

**Steps to reproduce the bug**

Create a suite and add `_ = " "` to the environment section for any task.

**Expected behavior**

Resultant job script to have `_=" "`

**Screenshots**

N/A

**Additional context**

Not a common need, but it should be allowed.

**Comment by oliver-sanders** [comment]
This whitespace stripping is a parsec feature (the configuration file parser Cylc uses), I believe it is applied to all settings.

I don't think we can change this as a large number of workflows are relying on this behaviour to strip arbitrary whitespace from environment variables.

This is also used to allow environment variable definitions (which may be sub-shells including line breaks) to be written over multiple lines e.g:

```
[environment]
    FOO = """
        bar
    """
```

**Comment by ColemanTom** [comment]
It would only be if the variable is enclosed in quotes, like in my example that I think it would make sense. In my view, if you have trailing spaces and have intentionally put double or single quotes around it, it must be intentional.

**Comment by hjoliver** [comment]
> I don't think we can change this as a large number of workflows are relying on this behaviour to strip arbitrary whitespace from environment variables.

Presumably you mean this sort of thing (using "_" to mean whitespace):
```
DRINK =___beer________
```
where the value should end up as `beer` (no spaces).

I think it's not so difficult because we can distinguish quoted values.   I'll put up a small PR for consideration...



 I'll post a small PR for consideration.

**Comment by oliver-sanders** [comment]
TLDR; The current pattern of stripping leading/trailing whitespace from env vars goes back a long way, we now have a substantial body of workflows which have grown into this pattern, for several of them, changing tack now would result in breaking changes.

Removing the bug label as this is an intentional design choice (will try do do a better job of documenting this).

The workarounds are:

* Use escape characters where possible.
* Refactor script logic.
* Set the environment variable within a `*script` setting.

Leaving this issue open for now, should close going forward unless there is further demand for this functionality.
